### ISSUE TITLE
HMR Issue with Parcel 2.0.0 release

### ISSUE THREAD

**Issue by lucaseverett** [solidjs/solid#693]
**Describe the bug**
When using Parcel 2.0.0 final release, the root node is duplicated upon saving a file, rather than replacing it. For example, when using the Solid Parcel Starter, you'll see the following after saving a file.

```
Solid Parcel Starter

Solid Parcel Starter
```

**To Reproduce**
1. Download Solid Parcel Starter from https://gitlab.com/enom/solid-parcel-starter
2. `npm i` and `npm run dev`
3. Make a change to index.js and observe that the page refreshes in the browser as expected.
4. Stop server, then update parcel in package.json from "next" to "2.0.0."
5. `npm i` and `npm run dev`
6. Make a change to index.js and observe that the root node is duplicated rather than replaced

**Expected behavior**
Upon saving a file the page content should be replaced rather than duplicated.

**Additional context**
I'm not sure if this is a Parcel bug or something within Solid. It doesn't happen when using Parcel 2.0.0-rc.0. The default behavior of Parcel is to simply reload a page if HMR isn't explicitly called.

**Comment by ryansolid** [comment]
@lucaseverett Did this ever go anywhere? Did you create the issue with Parcel?

**Comment by lucaseverett** [comment]
I created an issue for it, but there hasn't been any action on it yet.

https://github.com/parcel-bundler/parcel/issues/7171

**Comment by LXSMNSYC** [comment]
Must be an upstream issue. I tested the latest Parcel 2 with solid and solid-refresh@^0.3.0 and there shouldn't be any issues.

**Comment by lucaseverett** [comment]
Yes, it's a Parcel bug, and it was caused by a [commit made on September 25](https://github.com/parcel-bundler/parcel/blame/e0440dcbe6d6e419a412c8c568d94715cdb4282a/packages/runtimes/hmr/src/loaders/hmr-runtime.js#L339). There are several open issues that all seem to be related to this bug.

I edited the hmr-runtime.js file and removed `return true` from the code in the commit referenced above, and that indeed fixed the issue. I don't know enough about how the hmr-runtime works to know what needs to be rewritten. At least the cause is known, though.

https://github.com/parcel-bundler/parcel/issues/7307
